src/utils/daemon.js

```diff
@@ -1,63 +1,104 @@
 import IPFSFactory from 'ipfsd-ctl'
 import logger from './logger'
+import { showRepoApiFileErrorMessage } from './errors'
+import { join } from 'path'
+import fs from 'fs-extra'
+import { spawnSync } from 'child_process'
+import findExecutable from 'ipfsd-ctl/src/utils/find-ipfs-executable'
 
-export default async function createDaemon (opts) {
-  opts.type = opts.type || 'go'
-  opts.path = opts.path || ''
-  opts.flags = opts.flags || ['--migrate=true', '--routing=dhtclient']
-  opts.keysize = opts.keysize || 4096
+function repoFsck (path) {
+  const exec = findExecutable('go', join(__dirname, '..'))
+  spawnSync(exec, ['repo', 'fsck'], {
+    env: {
+      ...process.env,
+      IPFS_PATH: path
+    }
+  })
+}
 
-  if (opts.type !== 'go') {
-    throw new Error(`${opts.type} connection is not supported yet`)
+async function configure (ipfsd) {
+  const cfgFile = join(ipfsd.repoPath, 'config')
+  const cfg = await fs.readJSON(cfgFile)
+
+  let origins = []
+  try {
+    origins = cfg.API.HTTPHeaders['Access-Control-Allow-Origin']
+  } catch (e) {
+    logger.warn(e)
   }
 
-  const factory = IPFSFactory.create({ type: opts.type })
+  if (!Array.isArray(origins)) {
+    origins = []
+  }
+
+  if (!origins.includes('webui://-')) origins.push('webui://-')
+  if (!origins.includes('https://webui.ipfs.io')) origins.push('https://webui.ipfs.io')
+
+  cfg.API.HTTPHeaders['Access-Control-Allow-Origin'] = origins
+  cfg.API.HTTPHeaders['Access-Control-Allow-Methods'] = ['PUT', 'GET', 'POST']
+
+  await fs.writeJSON(cfgFile, cfg)
+}
+
+async function spawn ({ type, path, keysize = 0 }) {
+  const factory = IPFSFactory.create({ type: type })
 
-  const ipfsd = await new Promise((resolve, reject) => {
+  return new Promise((resolve, reject) => {
     factory.spawn({
       disposable: false,
       defaultAddrs: true,
-      repoPath: opts.path
+      repoPath: path
     }, (e, ipfsd) => {
       if (e) return reject(e)
       if (ipfsd.initialized) {
         return resolve(ipfsd)
       }
 
       ipfsd.init({
-        directory: opts.path,
-        keysize: opts.keysize
+        directory: path,
+        keysize: keysize
       }, e => {
         if (e) return reject(e)
         resolve(ipfsd)
       })
     })
   })
+}
 
-  if (!ipfsd.started) {
-    await new Promise((resolve, reject) => {
-      ipfsd.start(opts.flags, err => {
-        if (err) {
-          return reject(err)
-        }
+async function start (ipfsd, { flags }) {
+  await new Promise((resolve, reject) => {
+    ipfsd.start(flags, err => {
+      if (err) {
+        return reject(err)
+      }
 
-        resolve()
-      })
+      resolve()
     })
+  })
+}
+
+export default async function (opts) {
+  const ipfsd = await spawn(opts)
+  await configure(ipfsd)
```

Comment:
If ipfsd connects to an existing daemon process, and we change it's config, we would need to restart that process for the config to be applied.

Reply:
@olizilla we're now editing the config file directly so we won't even be able to change the config of remote daemons. Perhaps it would be better to revert this change and use the API.

Reply:
Ping @olizilla

Reply:
@olizilla I'm merging this so we can speed up dev. This piece of code will be removed on #736 so let's not take too much time here.